internal/plan/plan.go

```diff
@@ -184,6 +184,59 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// ParseCommand returns a service command as two stream of strings.
+// The base command is returned as a stream and the default arguments
+// in [ ... ] group is returned as another stream.
+func (s *Service) ParseCommand() (base, extra []string, err error) {
+	args, err := shlex.Split(s.Command)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var inBrackets, gotBrackets bool
+
+	for idx, arg := range args {
+		if inBrackets {
+			if arg == "[" {
+				return nil, nil, fmt.Errorf("cannot nest [ ... ] groups")
+			}
+			if arg == "]" {
+				inBrackets = false
+				continue
+			}
+			extra = append(extra, arg)
+			continue
+		}
+		if gotBrackets {
+			return nil, nil, fmt.Errorf("cannot have any args after [ ... ] group")
+		}
+		if arg == "[" {
+			if idx == 0 {
+				return nil, nil, fmt.Errorf("cannot have [ ... ] group as prefix")
```

Comment:
I would say just `"command cannot be ["` here.

However, note:
```
% whereis -b [ 
[: /usr/bin/[ /opt/toybox/bin/[
```
But I think we can just rule out support for `/usr/bin/[`. :-)

Reply:
Technically, `/usr/bin/[` will still be supported if `[` by itself is not an argument. `[5 -gt 4]`, `/usr/bin/[ 5 -gt 4]`, `[[ 5 > 4]]` will all work. What won't work is `[ 5 -gt 4 ]`.

That being said, I think the much more frequent case would be people adding all their commands inside `[ ... ]` to make it all overridable rather than people using `[` command. So I think the "cannot have [ ... ] group as prefix" error would be more relevant? Let me know if you think otherwise and I will change so.

Reply:
So, in your examples, only `/usr/bin/[ 5 -gt 4]` would work. `[5 -gt 4]` and `[[ 5 > 4 ]]` would look for `[5` and `[[` executable in `$PATH` respectively, which most likely doesn't exist. However, we can just rule out `[` as a service. :-) (Or suggest `test` as alternative.)

I find the message a bit confusing but maybe it's just me, so it's up to you (and others).

Reply:
a less technical but more accurate message would be: `command cannot be started with default arguments`

Reply:
Updated, thanks.